packages/cherry-pick-bot/src/cherry-pick.ts

```diff
@@ -0,0 +1,355 @@
+ * @param {OctokitType} octokit An authenticated Octokit instance
+ * @param {string} owner The repository owner
+ * @param {string} repo The repository name
+ * @param {string[]} commits The commit SHAs to cherry-pick
+ * @param {string} targetBranch The target branch of the pull request
+ * @returns {PullRequest}
+ */
+export async function cherryPickAsPullRequest(
+  octokit: OctokitType,
+  owner: string,
+  repo: string,
+  commits: string[],
+  targetBranch: string
+): Promise<PullRequest> {
+  logger.info(`cherry-pick ${commits} to ${targetBranch} via pull request`);
+  const hash = crypto.createHash('md5').update(commits.join(',')).digest('hex');
+  const newBranchName = `cherry-pick-${hash.substring(0, 6)}-${targetBranch}`;
+  const targetBranchHead = (
+    await octokit.repos.getBranch({
+      owner,
+      repo,
+      branch: targetBranch,
+    })
+  ).data.commit.sha;
+
+  logger.debug(`creating new branch: ${newBranchName} from ${targetBranch}`);
+  await createOrUpdateRef(
+    octokit,
+    owner,
+    repo,
+    newBranchName,
+    targetBranchHead
+  );
+
+  const newHeadSha = await cherryPickCommits(
+    octokit,
+    owner,
+    repo,
+    commits,
+    newBranchName
+  );
+  logger.debug(`cherry-picked as ${newHeadSha}`);
+
+  logger.debug(`opening pull request from ${newBranchName} to ${targetBranch}`);
+  const {data: pullRequest} = await octokit.pulls.create({
+    owner,
+    repo,
+    head: newBranchName,
+    base: targetBranch,
+    title: `chore: cherry-pick commit ${commits.join(', ')} to ${targetBranch}`,
+  });
+  return pullRequest;
+}
+
+/**
+ * Cherry-pick a commit to a specific branch
+ * @param {OctokitType} octokit An authenticated Octokit instance
+ * @param {string} owner The repository owner
+ * @param {string} repo The repository name
+ * @param {string[]} commits The commit SHAs to cherry-pick
+ * @param {string} targetBranch The target branch of the pull request
+ */
+export async function cherryPickCommits(
+  octokit: OctokitType,
+  owner: string,
+  repo: string,
+  commits: string[],
+  targetBranch: string
+): Promise<string> {
+  logger.info(`cherry-pick ${commits} to branch ${targetBranch}`);
+
+  logger.debug(`fetching sha for ref ${targetBranch}`);
+  const {
+    data: {
+      object: {sha: initialHeadSha},
+    },
+  } = await octokit.git.getRef({
+    owner,
+    repo,
+    ref: `heads/${targetBranch}`,
+  });
+
+  const temporaryRefName = `temp-${targetBranch}`;
+  logger.debug(`creating temporary ref: ${temporaryRefName}`);
+  await createOrUpdateRef(
+    octokit,
+    owner,
+    repo,
+    temporaryRefName,
+    initialHeadSha
+  );
+
+  logger.debug(`fetching ${targetBranch} tree SHA`);
+  const {
+    data: {
+      tree: {sha: initialHeadTree},
+    },
+  } = await octokit.git.getCommit({
+    owner,
+    repo,
+    commit_sha: initialHeadSha,
+  });
+
+  let headSha = initialHeadSha;
+  let headTree = initialHeadTree;
+
+  for (const sha of commits) {
```

Comment:
nit: I think this loop might read better if pulled into a helper method `findHeadSha(intiailHeadSha)`.

That way you don't have to use the temporary variables, which are set kind of late in the algorithm.